client-ts/signalr/src/Transports.ts

```diff
@@ -37,28 +38,36 @@ export class WebSocketTransport implements ITransport {
         this.accessTokenFactory = accessTokenFactory || (() => null);
     }
 
-    public connect(url: string, requestedTransferMode: TransferMode, connection: IConnection): Promise<TransferMode> {
+    public connect(url: string, transferFormat: TransferFormat, connection: IConnection): Promise<void> {
+        Arg.isRequired(url, "url");
```

Comment:
Is this idiomatic? This looks too .net ish

Reply:
Does it matter though? It's internal to the client

Reply:
I couldn't find a consistent idiomatic way to do this. Some people use argument validator methods like this. There are some tricks you can do with default parameter values (`url: string = someFuncThatThrowsIfExecuted()`) but those only work when the parameter is `undefined`, a `null` value would be accepted.

I think we just need to put something in and start doing it.

Reply:
> Does it matter though? It's internal to the client

Yes, it matters.